doc/ExpertOptions.md

```diff
@@ -43,33 +43,45 @@ Make sure you've installed the dx-toolkit CLI, and initialized it with `dx login
+This compiles `foo.wdl` to platform workflow `foo` in specified dx's project and folder (defaults to currently selected project and '/'). The generated workflow can then be run as usual using `dx run`. For example, if the workflow takes string argument `X`, then: ``` dx run foo -i0.X="hello world" ```
 
 Compilation can be controled with several parameters.
 
 | Option   |  Description |
 | ------   | ------------ |
-| archive  | Archive older versions of applets and workflows |
-| defaults | A file with default parameter settings. The syntax is Cromwell style. |
-| destination | Set the output folder on the platform |
-| execTree | Prints a JSON or text representation of the compiled workflow |
-| extras   | JSON formatted file with additional options |
-| force    | Overwrite existing applets/workflows if they have changed |
-| inputs   | A cromwell style inputs file |
-| imports  | Directory to search for imported WDL files |
-| locked   | Create a locked-down workflow |
+| archive  | Archive older versions of applets.|
+| compileMode \<string\> | Compilation mode - a debugging flag for internal use.|
+| defaults \<string\> | JSON file with standard-formatted default values. |
+| defaultInstanceType \<string\> | The default instance type to use for "helper" applets that perform runtime evaluation of instance type requirements. This instance type is also used when the '-instanceTypeSelection dynamic' option is set. This value is overriden by any defaults set in extras. |
+| destination \<string\> | Full platform path (project:/folder) |
+| execTree \[json,pretty\] | Print a JSON representation of the workflow. |
+| extras \<string\> | JSON file with extra options (see documentation). |
+| inputs \<string\> | JSON file with standard-formatted input values. May be specified multiple times. A DNAnexus JSON input file is generated for each standard input file. |
+| instanceTypeSelection \[static,dynamic\] |  Whether to attempt to select instance types at compile time or tasks with runtime requirements that can all be statically valuated (static, the default), or to defer all instance type election to runtime (dynamic). Using static instance type election can save time and cost, but it requires the user(s) unning the applet/workflow to have access to the same instance ypes as the user who compiled it.
+| imports \<string\> | Directory to search for imported WDL files. May be specified multiple times. |
+| locked   | Create a locked workflow. When running a locked workflow, input values may only be specified for the top-level workflow. |
+| leaveWorkflowsOpen | Leave created workflows open (otherwise they are closed). |
 | noListings | CWL-specific option to prevent full folder listings in generated input files |
```

Comment:
the `noListings` option doesn't seem to be implemented. However it is described in Line 250. Is there anything I had missed? @jdidion @r-i-v-a

Reply:
I would just copy here what is currently in compiler/src/main/scala/dxCompiler/Main.scala usageMessage on `develop` branch, which does not currently include noListings.

But I see it is also mentioned here
"CWL does provide a mechanism for ensuring reproducibility of jobs that take directory inputs, via the `listing` field. We strongly recommend that CWL users specify the folder listing for each directory input. A job will only be reused if both the folder and the listing are identical. The ordering of the listing is taken into consideration when making the comparison, so the listing must be generated deterministically. The default behavior of dxCompiler when using the `-input` option is to generate input files with full listings for all directories, unless the `-noListings` option is specified. An example of a folder with a listing is:"

I think it should not be mentioned in the documentation if not implemented yet; once it is implemented, we can update both the usageMessage and the docs. @jdidion